File: Codes in Python3/Array/Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxPro = 0
        minPrice = float('inf')
        for i in range(len(prices)):
            minPrice = min(minPrice, prices[i])
            maxPro = max(maxPro, prices[i] - minPrice)
        return maxPro

File: Codes in C++/Array/Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
      int n=prices.size();
      int buy=prices[0];
      int sell=0;
      for(int i=0;i<n;i++)
      {
          buy=min(buy,prices[i]);
          sell=max(sell,prices[i]-buy);
      }
     /*vector<int> diff(n-1,1);
      int currSum=0;int maxSum=0;
      for(int i=0;i<n-1;i++)
      {
          diff[i]=prices[i+1]-prices[i];
      }
     //Kadane's Algo
     int startIndex=0;
     for(int i=0;i<n-1;i++)
     {
        if(currSum+diff[i]>0)
            currSum+=diff[i];
        else currSum=0;
         if(currSum>maxSum)
         maxSum=currSum;
     }*/
      return sell;
    }
};